utils/notification_service_quantization.py

```diff
@@ -0,0 +1,258 @@
+
+        if self.n_failures > 0:
+            blocks.append(self.failures_overwiew)
+            blocks.append(self.failures_detailed)
+
+        if self.n_failures == 0:
+            blocks.append(self.no_failures)
+
+        return json.dumps(blocks)
+
+    @property
+    def time(self) -> str:
+        all_results = self.results.values()
+        time_spent = []
+        for r in all_results:
+            if len(r["time_spent"]):
+                time_spent.extend([x for x in r["time_spent"].split(", ") if len(x.strip())])
+        total_secs = 0
+
+        for time in time_spent:
+            time_parts = time.split(":")
+
+            # Time can be formatted as xx:xx:xx, as .xx, or as x.xx if the time spent was less than a minute.
+            if len(time_parts) == 1:
+                time_parts = [0, 0, time_parts[0]]
+
+            hours, minutes, seconds = int(time_parts[0]), int(time_parts[1]), float(time_parts[2])
+            total_secs += hours * 3600 + minutes * 60 + seconds
+
+        hours, minutes, seconds = total_secs // 3600, (total_secs % 3600) // 60, total_secs % 60
+        return f"{int(hours)}h{int(minutes)}m{int(seconds)}s"
+
+    @property
+    def failures_overwiew(self) -> Dict:
+        return {
+            "type": "section",
+            "text": {
+                "type": "plain_text",
+                "text": (
+                    f"There were {self.n_failures} failures, out of {self.n_tests} tests.\n"
+                    f"The suite ran in {self.time}."
+                ),
+                "emoji": True,
+            },
+            "accessory": {
+                "type": "button",
+                "text": {"type": "plain_text", "text": "Check Action results", "emoji": True},
+                "url": f"https://github.com/huggingface/transformers/actions/runs/{os.environ['GITHUB_RUN_ID']}",
+            },
+        }
+
+    @property
+    def failures_detailed(self) -> Dict:
+        failures = {k: v["failed"] for k, v in self.results.items()}
+
+        individual_reports = []
+        for key, value in failures.items():
+            device_report = self.get_device_report(value)
+            if sum(value.values()):
+                report = f"{device_report}{key}"
+                individual_reports.append(report)
+
+        header = "Single |  Multi | Category\n"
+        failures_report = prepare_reports(
+            title="The following quantization tests had failures", header=header, reports=individual_reports
+        )
+
+        return {"type": "section", "text": {"type": "mrkdwn", "text": failures_report}}
+
+    def post(self):
+        payload = self.payload
+        print("Sending the following payload")
+        print(json.dumps({"blocks": json.loads(payload)}))
+
+        text = f"{self.n_failures} failures out of {self.n_tests} tests," if self.n_failures else "All tests passed."
+
+        self.thread_ts = client.chat_postMessage(
+            channel="#transformers-ci-daily-quantization",
```

Comment:
well, I didn't know this works. Nice!

Reply:
Yes I learned this from the peft/accelerate CI log report [here](https://github.com/huggingface/peft/blob/02b5aeddf9c1ea11451f10a8a26da7e5df8cca4a/scripts/log_reports.py#L139).